avocado/core/task/statemachine.py

```diff
@@ -74,20 +75,69 @@ async def bootstrap(self):
-        if requirements_ok:
+        # check for requirements a task may have
+        requirements_ok = (
+                await self._spawner.check_task_requirements(runtime_task))
+        if not requirements_ok:
             async with self._state_machine.lock:
-                self._state_machine.ready.append(runtime_task)
-        else:
+                fail_on_triage(runtime_task)
+            return
+
+        # handle task dependencies
+        if runtime_task.task.dependencies:
+            # check of all the dependency tasks finished
             async with self._state_machine.lock:
-                self._state_machine.finished.append(runtime_task)
-                runtime_task.status = 'FAILED ON TRIAGE'
+                finished_tasks = [rt_task.task for rt_task
+                                  in self._state_machine.finished]
+            if not runtime_task.task.dependencies.issubset(finished_tasks):
+                async with self._state_machine.lock:
+                    self._state_machine.triaging.append(runtime_task)
+                    runtime_task.status = 'WAITING DEPENDENCIES'
+                await asyncio.sleep(0.1)
+                return
+
+            # dependencies finished, let's check if they finished
+            # successfully, so we can move on with the parent task
+            for task in runtime_task.task.dependencies:
+                # check if this dependency `task` failed on triage
+                # this is needed because this kind of task fail does not
+                # have information in the status repo
+                for finished_rt_task in self._state_machine.finished:
+                    if finished_rt_task.task is task:
+                        if finished_rt_task.status == 'FAILED ON TRIAGE':
+                            async with self._state_machine.lock:
+                                fail_on_triage(runtime_task)
+                            return
+                # from here, this dependency `task` ran, so, let's check
+                # the latest data it has in the status repo
+                latest_task_data = \
+                    self._state_machine._status_repo.get_latest_task_data(
+                        str(task.identifier))
+                # maybe, the latest data is not available yet
+                if latest_task_data is None:
+                    async with self._state_machine.lock:
+                        self._state_machine.triaging.append(runtime_task)
+                    await asyncio.sleep(0.1)
+                    return
+                # if this dependency task failed, skip the parent task
+                if latest_task_data['result'] not in ['pass']:
+                    async with self._state_machine.lock:
+                        fail_on_triage(runtime_task)
+                    return
+
+        # the task is ready to run
+        async with self._state_machine.lock:
+            self._state_machine.ready.append(runtime_task)
```

Comment:
This method is too big, do you think would be easy to split into small ones?

Reply:
The method is 64 lines long; removing the comments and blank lines makes it 45 lines. The blocks that update the state machine queue use 2 or 3 lines because we need to lock the state machine queue, and we have 6 occurrences of them. Personally, I don't see a need to split because most of the code is "accessory", but if it helps in readability, I can move lines 109-136 to a new function.

Reply:
Yes, big is relative, but 64 lines are too big IMO. I try to follow the rule of 30". :) And I appreciate this small split. Thank you.